client/src/js/users/components/__tests__/Detail.test.js

```diff
@@ -0,0 +1,139 @@
+import { UserDetail, mapStateToProps, mapDispatchToProps } from "../Detail";
+
+describe("<UserDetail />", () => {
+    let props;
+
+    beforeEach(() => {
+        props = {
+            onGetUser: jest.fn(),
+            match: {
+                params: {
+                    userId: "foo"
+                }
+            },
+            groupsFetched: true,
+            onListGroups: jest.fn(),
+            onSetUserRole: jest.fn(),
+            detail: {
+                id: "foo",
+                groups: "bar",
+                administrator: "baz",
+                identicon: "foo",
+                primary_group: "Bar",
+                permissions: { foo: "bar" }
+            },
+            onRemoveUser: jest.fn(),
+            error: []
+        };
+    });
+
+    it("should render", () => {
+        const wrapper = shallow(<UserDetail {...props} />);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it("componentDidMount should call onGetUser", () => {
+        expect(props.onGetUser).not.toHaveBeenCalled();
+        const wrapper = shallow(<UserDetail {...props} />);
+        expect(props.onGetUser).toHaveBeenCalledWith("foo");
+    });
+
+    it("componentDidMount should call onListGroups", () => {
+        const wrapper = shallow(<UserDetail {...props} />);
+        expect(props.onListGroups).not.toHaveBeenCalled();
+    });
+    it("componentDidMount should not call onListGroups when [props.groupsFetched = false]", () => {
+        props.groupsFetched = false;
+        const wrapper = shallow(<UserDetail {...props} />);
+        expect(props.onListGroups).toHaveBeenCalled();
+    });
+});
+
+describe("mapStateToProps", () => {
+    const state = {
+        users: {
+            detail: "foo"
+        },
+        account: {
+            id: "foo",
+            administrator: true
+        },
+        groups: {
+            list: "foo",
+            fetched: true
+        }
+    };
+
+    const props = mapStateToProps(state);
+    expect(props).toEqual({
+        detail: "foo",
+        activeUser: "foo",
+        activeUserIsAdmin: true,
+        groups: "foo",
+        groupsFetched: true,
+        error: ""
+    });
+});
+
+describe("mapDispatchToProps", () => {
+    let dispatch;
+    let result;
+
+    beforeEach(() => {
+        dispatch = jest.fn();
+        result = mapDispatchToProps(dispatch);
+    });
+
+    it("should return onGetUser() in props", () => {
+        const userId = "foo";
+        result.onGetUser(userId);
+        expect(dispatch).toHaveBeenCalledWith({
+            type: "GET_USER_REQUESTED",
+            userId
+        });
+    });
+    it("should return onRemoveUser() in props", () => {
+        const userId = "foo";
+        result.onRemoveUser(userId);
+        expect(dispatch).toHaveBeenCalledWith({
+            type: "REMOVE_USER_REQUESTED",
+            userId
+        });
+    });
+    it("should return onClose() in props", () => {
+        const path = "/administration/users";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['path' is assigned a value but never used. (no-unused-vars)](https://app.codacy.com/app/virtool/virtool/pullRequest?prid=4503767)